kcqrs-core/src/main/kotlin/com/clouway/kcqrs/core/Configuration.kt

```diff
@@ -0,0 +1,13 @@
+package com.clouway.kcqrs.core
+
+/**
+ * Configuration is representing the configuration of the KCQRS.
+ *
+ * @author Miroslav Genov (miroslav.genov@clouway.com)
+ */
+interface Configuration {
+    /**
+     * Determines the topic name of the AggregateRoot.
+     */
+    fun topicName(aggregate: AggregateRoot): String
```

Comment:
I think we don't need to pass the whole aggregate to fetch the topic. Do you think if we pass just ``` class<T> ``` will be enough to fetch the topic ???

Reply:
In most cases this method will return a hardcoded value of the topic name and it will be in the entry point of the micro-service. If micro-services are granular then each one will use a single topic. 

The instance may provide and additional information then the class as it contains the aggregateId version and etc. Also the check for a certain type in kotlin would be something like:

```kotlin
if (aggregate is Invoice) {
  return "invoicing"
}
```
which looks simple enough.